src/ansys/fluent/core/utils/search.py

```diff
@@ -224,10 +248,431 @@ def inner(tree, path, root_path):
+        api_tree_file.touch()
+        with open(api_tree_file, "w") as file:
+            for api_object in api_objects:
+                file.write(f"{api_object}")
+                file.write("\n")
+
+    text_file_folder = Path(os.path.join(pyfluent.CODEGEN_OUTDIR, "api_tree"))
+    text_file_folder.mkdir(parents=True, exist_ok=True)
+
+    _write_api_tree_file(api_objects=api_objects, text=True)
+    _write_api_tree_file(api_objects=api_tui_objects, tui=True)
+    _write_api_tree_file(api_objects=api_object_names, name=True)
+
+
+def _get_api_object_names():
+    """Get API object names."""
+    return [
+        line.rstrip("\n")
+        for line in open(get_api_tree_file_name(name=True), "r", encoding="utf-8")
+    ]
+
+
+def _print_search_results(queries: list):
+    """Print search results.
+
+    Parameters
+    ----------
+    queries: list
+        List of search string to match API object names.
+    """
+    api_tree = get_api_tree_file_name(text=True)
+    api_tui_tree = get_api_tree_file_name(tui=True)
+    text_files = [api_tree, api_tui_tree]
+    for text_file in text_files:
+        api_objects = [
+            line.rstrip("\n") for line in open(text_file, "r", encoding="utf-8")
+        ]
+        for query in queries:
+            for api_object in api_objects:
+                if query in api_object:
+                    print(api_object)
+
+
+def _get_wildcard_matches_for_word_from_names(word: str, names: list):
+    """Get wildcard matches for the given word.
+
+    Parameters
+    ----------
+    word: str
+       Word to search for.
+    names: list
+        All API object names.
+
+    Returns
+    -------
+    wildcard_matches: list
+        Matched API object names.
+    """
+    wildcard_matches = [name for name in names if fnmatch.fnmatch(name, word)]
+    valid_wildcard_matches = [
+        wildcard_match for wildcard_match in wildcard_matches if wildcard_match in names
+    ]
+    return valid_wildcard_matches
+
+
+def _search_wildcard(search_string: str):
+    """Perform wildcard search for a word through the Fluent's object hierarchy.
+
+    Parameters
+    ----------
+    search_string: str
+        Word to search for. Semantic search is default.
+
+    Returns
+    -------
+        List of search string matches.
+    """
+    names = _get_api_object_names()
+    queries = _get_wildcard_matches_for_word_from_names(search_string, names)
+    if queries:
+        _print_search_results(queries)
+
+
+def _get_exact_match_for_word_from_names(
+    word: str,
+    names: list,
+):
+    """Get exact match for the given word.
+
+    Parameters
+    ----------
+    word: str
+        Word to search for.
+    names: list
+        All API object names.
+
+    Returns
+    -------
+    exact_matches: list
+        List of exact match.
+    """
+    return [name for name in names if word == name]
+
+
+def _get_match_case_for_word_from_names(
+    word: str,
+    names: list,
+):
+    """Get match case for the given word.
```

Comment:
Again, the doc string does not clarify what this is. Very confusing.

Reply:
There is a difference between definitions of match_case for VS code search and Google find text. I referred definition of Google which is inconsistent with ours existing _search function results. it's corrected now. Thank you.